### ISSUE TITLE
Centre of rotation

### ISSUE THREAD

**Issue by GurveerSohal** [naurril/SUSTechPOINTS#164]
Is there a way to change the centre of rotation for the loaded pointcloud?

If I have a large point cloud, it's hard to navigate that because the centre of rotation seems to be fixed. Also, if the coordinates of the pointcloud are far from the centre, it's hard to rotate too.

Is there anywhere in the code I can get started on changing this?

**Comment by naurril** [comment]
Did you try `reset view` in context menu?

**Comment by GurveerSohal** [comment]
> Did you try `reset view` in context menu?

Yes, but the centre of rotation seems to be fixed away from the point cloud itself. Large point clouds work fine if we change their coordinates to be centred around the centre of the circle ruler

**Comment by naurril** [comment]
'reset view' simply computes the centroid of points as the center, you can
modify it to meet your needs, search resetview in editor.js

**Comment by GurveerSohal** [comment]
I saw that, but is there anywhere you define the centre of rotation for the camera itself?

**Comment by naurril** [comment]
you can set it when the main camera is initialized

https://github.com/naurril/SUSTechPOINTS/blob/522d1daef1e829f433cada6f420189ab3daee249/public/js/view.js#L80

**Comment by GurveerSohal** [comment]
Thanks!

**Comment by GurveerSohal** [comment]
When you pan using right click, it changes the centre of rotation (we checked using a console log in view.render()). I saw that the right click event is handled in handle_right_click() in editor.js, but I don't see where the drag event is handled. Can you please tell me where it is?

**Comment by naurril** [comment]
the camera operations including `pan`  are handled by `OrbitControls`

**Comment by GurveerSohal** [comment]
Thanks! We found that ussing the following:
`this.viewManager.mainView.orbit.target.set(pos.x, pos.y, center.z);`
in resetView() results in the desired behaviour for changing the center of rotation. (Changing the z value from 0 to center.z).
However, using any of the following:
`this.viewManager.mainView.orbit.target.set(center.x, center.y, center.z);`
`this.viewManager.mainView.orbit.target.set(pos.x, pos.y, pos.z);`
results in some problems, we are unsure of why yet.
